### PR TITLE
Replace moment and jstimezonedetect with dayjs

### PR DESCRIPTION
**Pull request by 4lp**
Replaced moment throughout the site, also figured it would further reduce the bundle size to replace jstimezonedetect with dayjs's `tz.guess()` method.

Let me know if anything looks amiss or if you'd like me to change anything!

### CODE REVIEW

**Comment by mcfiredrill** [comment]
One merge conflict here as well. @4lp 

Other than that looks great! 
I will try to confirm that the bundle size using this dropped since I'm curious.
https://github.com/kaliber5/ember-cli-bundle-analyzer#usage

**Comment by 4lp** [comment]
ok that should do it! let me know if there are any issues

File: package.json [hunk 1]
```diff
@@ -71,7 +72,6 @@
     "ember-lodash": "4.19.5",
     "ember-maybe-import-regenerator": "^0.1.6",
     "ember-modal-dialog": "^3.0.0-beta.4",
-    "ember-moment": "^8.0.0",
```
Comment (mcfiredrill) [inline package.json:74]: Can you also try removing `ember-cli-moment-shim` as well ? @4lp

**Comment by mcfiredrill** [comment]
Hmm I tried removing a few more packages that depend on moment:
https://github.com/datafruits/datafruits/commit/3c185e1ebd2435163e1f45477cd0e7a262b39fcf

But it was still in the bundle.

The main issue is that ember-calendar depends on it.
```
 tony@dokkebi  ~/src/datafruits   replace-moment-remove-pkgs  yarn why moment   
yarn why v1.22.4
[1/4] Why do we have the module "moment"...?
[2/4] Initialising dependency graph...
warning Resolution field "babel-plugin-ember-modules-api-polyfill@2.13.0" is incompatible with requested version "babel-plugin-ember-modules-api-polyfill@^2.13.4"
warning Resolution field "babel-plugin-ember-modules-api-polyfill@2.13.0" is incompatible with requested version "babel-plugin-ember-modules-api-polyfill@^3.2.0"
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "moment@2.26.0"
info Reasons this module exists
   - "ember-calendar#ember-cli-moment-shim" depends on it
   - Hoisted from "ember-calendar#ember-cli-moment-shim#moment"
   - Hoisted from "ember-calendar#ember-cli-moment-shim#moment-timezone#moment"
info Disk size without dependencies: "5MB"
info Disk size with unique dependencies: "5MB"
info Disk size with transitive dependencies: "5MB"
info Number of shared dependencies: 0
Done in 1.82s.
```

However honestly, I'm thinking of dumping ember-calendar for something simpler.
https://github.com/datafruits/datafruits/issues/555
Perhaps we can come back to this PR when that happens.

**Comment by 4lp** [comment]
> Hmm I tried removing a few more packages that depend on moment:
> [3c185e1](https://github.com/datafruits/datafruits/commit/3c185e1ebd2435163e1f45477cd0e7a262b39fcf)
> 
> But it was still in the bundle.
> 
> The main issue is that ember-calendar depends on it.
> 
> ```
>  tony@dokkebi  ~/src/datafruits   replace-moment-remove-pkgs  yarn why moment   
> yarn why v1.22.4
> [1/4] Why do we have the module "moment"...?
> [2/4] Initialising dependency graph...
> warning Resolution field "babel-plugin-ember-modules-api-polyfill@2.13.0" is incompatible with requested version "babel-plugin-ember-modules-api-polyfill@^2.13.4"
> warning Resolution field "babel-plugin-ember-modules-api-polyfill@2.13.0" is incompatible with requested version "babel-plugin-ember-modules-api-polyfill@^3.2.0"
> [3/4] Finding dependency...
> [4/4] Calculating file sizes...
> => Found "moment@2.26.0"
> info Reasons this module exists
>    - "ember-calendar#ember-cli-moment-shim" depends on it
>    - Hoisted from "ember-calendar#ember-cli-moment-shim#moment"
>    - Hoisted from "ember-calendar#ember-cli-moment-shim#moment-timezone#moment"
> info Disk size without dependencies: "5MB"
> info Disk size with unique dependencies: "5MB"
> info Disk size with transitive dependencies: "5MB"
> info Number of shared dependencies: 0
> Done in 1.82s.
> ```
> 
> However honestly, I'm thinking of dumping ember-calendar for something simpler.
> #555
> Perhaps we can come back to this PR when that happens.

Ah dang, I was hoping that Ember could at least use code splitting to only include moment in the bundle for the timetable route but it looks like Ember hasn't implemented that yet : / 

In that case, we can totally shelve this for later! Thanks for looking at it : )

**Comment by mcfiredrill** [comment]
>Ah dang, I was hoping that Ember could at least use code splitting to only include moment in the bundle for the timetable route but it looks like Ember hasn't implemented that yet : /

Yeah no code splitting in Ember afaik. :(

I merged the timetable rewrite, https://github.com/datafruits/datafruits/issues/555
So ember-calendar is out of the picture now.

Haven't checked yet but that should remove all moment deps.

**Comment by mcfiredrill** [comment]
Tried to fix conflicts and update the PR here https://github.com/datafruits/datafruits/pull/574

**Comment by mcfiredrill** [comment]
closing in favor of #574

**Pull request closed without merging**